src/Grid.cpp

```diff
@@ -89,31 +83,23 @@ void Cell::update_position() {
 
 // Grid
 
-Grid::Grid() {
-    
-    grid_width = 0;
-    grid_height = 0;
-    spacing = DEFAULT_SPACING;
-    
-}
-
-Grid::Grid(uint32_t width, uint32_t height) {
-    
-    grid_width = width;
-    grid_height = height;
-    spacing = DEFAULT_SPACING;
+Grid::Grid(uint32_t width, uint32_t height) : grid_width(width), grid_height(height) {
     
     initialize();
     
 }
 
 void Grid::random_init() {
-    if(!cells.empty()) {
-        for(auto& cell : cells) {
-            int r = floor(rand()*100);
-            if(r > 40) {
-                cell.state = CELL_STATE::CELL_ON;
-            }
+    
+    //generate random number with a uniform distribution
+    std::random_device device;
+    std::mt19937 rng(device());
```

Comment:
This isn't correct, you need a persistent engine across multiple calls. Try using the snippet I'd shared, or I can post another one too.

Reply:
not sure which snippet you're referring to, I might have missed it. I guess I should really be doing this in main, and using some new `activate_cells()` function. because it doesn't really feel right for `Grid` to own this method

Reply:
Let me just write a new snippet instead:
```cpp
int random_range(int lo, int hi) {
  static auto engine = std::default_random_engine{std::random_device{}()};
  return std::uniform_int_distribution<int>{lo, hi}(engine);
}
```